src/routes/Edit.svelte

```diff
@@ -14,14 +14,48 @@
     onMount(async () => {
         ga("send", "pageview");
         ga("send", "event", "version", mermaidVersion, mermaidVersion);
-        fromUrl(params.data);
+
+        const key = "historyList_key";
+        historyList = JSON.parse(localStorage.getItem(key) || "[]");
+        let hisCode = historyList.length > 0 ? historyList[historyList.length - 1] : null;
+
+        if (params.data) {
+            fromUrl(params.data);
+        } else if (hisCode) {
+            updateCodeStore({
+                code: hisCode.code,
+                mermaid: { theme: "default" },
+                updateEditor: true,
+            });
+        }
+
+        let code = null;
+        codeStore.subscribe( state => {
+            code = state && state.code || code;
+        });
+        
+        setInterval(() => {
+            if (code != hisCode) {
+                //save history
+                historyList.push({
+                    time: new Date().toISOString(),
+                    code: hisCode = code
+                });
+                if (historyList.length > 10) {
+                    historyList.shift();
+                }
+                historyList = historyList; //triggered update
+                localStorage.setItem(key, JSON.stringify(historyList));
+            }
+        }, 1 * 60 * 1000);
     });
     // export let code = '';
     // export let classes = '';
     // export let error = {};
     // export let token = '';
     // export let expected = '';
     export let params = {};
+    let historyList = [];
```

Comment:
It would be better to have this variable above the onMount function for readability.